internal/datadog/gates.go

```diff
@@ -0,0 +1,14 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package datadog // import "github.com/open-telemetry/opentelemetry-collector-contrib/internal/datadog"
+
+import "go.opentelemetry.io/collector/featuregate"
+
+var ReceiveResourceSpansV2FeatureGate = featuregate.GlobalRegistry().MustRegister(
```

Comment:
I would suggest moving this file to pkg/datadog then delete internal/datadog. Not worth adding a new module AFAICS